src/java.base/share/classes/jdk/internal/misc/CDS.java

```diff
@@ -82,9 +82,24 @@ public static boolean isDumpingStaticArchive() {
         return (configStatus & IS_DUMPING_STATIC_ARCHIVE) != 0;
     }
 
+    public static boolean isSingleThreadVM() {
+        return isDumpingStaticArchive();
+    }
+
     private static native int getCDSConfigStatus();
     private static native void logLambdaFormInvoker(String line);
 
+
+    private static ArrayList<Object> keepAliveList;
```

Comment:
Is it worth adding a comment stating this list should only be populated during an assembly phase?  That it should be null in both training and production runs?

Reply:
I added comments and asserts for clarification.